### PR TITLE
fix: windows code signing

### PR DESCRIPTION
**Pull request by filfreire**
## todo
- [x] disable winCodeSign unused download
- [x] remove unused CSC links for windows
- [x] try to get Update.exe and Insomnia.exe (stub) codesigned
- [x] prevent portable.exe windows (used for recurring builds) from being signed
- [x] test in a dummy GitHub actions workflow - https://github.com/Kong/insomnia/actions/runs/11363590563/artifacts/2062742332

## windows codesigning process after merging this PR

The approach is:

1) run `package:windows:unpacked` to bake `windows-unpacked` folder (which runs `npm run build:app`) with all common baked insomnia files
2) codesign the .dll and .exe unpacked files (common to all installers) (using GitHub actions intermediate steps that use [SSL.com's esigner-codesign](https://github.com/SSLcom/esigner-codesign)
3) run the `package:windows:dist` and point it to the same `windows-unpacked` folder (using electron-builder's `--dir` flag) for baking the temporary squirrel related files, signing with [SSL.com's ci-images](https://github.com/sslcom/ci-images/) via `customSign.js` hook, bake the installer package, and sign that as well from hook.

## debugging

found that Update.exe and Insomnia.exe (squirrel stub) are only available to codesign via hook

image

## Solution

We use the container image for EV codesigner from a custom-hook during packaging.

Afterwards, both the Stub, Updater.exe and the installer are all code-signed in a single go:
image

(using upstream branch instead of fork to test, closed https://github.com/Kong/insomnia/pull/8069)

### CODE REVIEW

**Review by gatzjames — DISMISSED** [review]

File: packages/insomnia/customSign.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const { exec } = require('child_process');
+const util = require('util');
+const path = require('path');
+const fs = require('fs');
+const execAsync = util.promisify(exec);
+
+exports.default = async function(configuration) {
```
Comment (jackkav) [inline packages/insomnia/customSign.js:10]: Can you add a comment indicating the prior art this was inspired by or copied from

File: packages/insomnia/electron-builder.config.js [hunk 2]
```diff
@@ -89,6 +89,10 @@ const config = {
         target: 'squirrel',
       },
     ],
+    signAndEditExecutable: false, // false to prevent unused winCodeSign download
+    sign: './customSign.js',
+    signingHashAlgorithms: ['sha256'], // avoid duplicate signing hook calls https://github.com/electron-userland/electron-builder/issues/3995#issuecomment-505725704
```
Comment (jackkav) [inline packages/insomnia/electron-builder.config.js:94]: But what about window 7 support 🙀 /s

**Review by jackkav — DISMISSED** [review]
maybe just add `npm run build:app &&` for consistency, can do it in future if you need

File: packages/insomnia/electron-builder.config.js (hunk 2 above)
Comment (filfreire) [reply packages/insomnia/electron-builder.config.js:94]: such is life

File: packages/insomnia/customSign.js (hunk 1 above)
Comment (filfreire) [reply packages/insomnia/customSign.js:10]: added

**Comment by filfreire** [comment]
@jackkav about:
> maybe just add npm run build:app && for consistency, can do it in future if you need

We should not add it in this case. The approach is:

- First we run `package:windows:unpacked` to bake `windows-unpacked` (which runs `npm run build:app`)
- Then we codesign the .dll and .exe unpacked files (common to all installers) 
- Then we run the `package:windows:dist` and point it to the same `windows-unpacked` folder  for baking the temporary squirrel related files, signing via customSign hook, bake the installer package, and sign that as well.


The unpacked files need to get codesigned in an intermediate step, and then we build the Squirrel package from the codesigned `windows-unpacked` folder for which we have already previously run `build:app`.

**Review by ryan-willis — APPROVED** [review]

**Pull request merged**